faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -196,18 +213,38 @@ class FaunaClient private (connection: Connection) {
       result.asInstanceOf[ArrayV].elems
     }
 
-  private def performRequest(body: JsonNode, timeout: Option[FiniteDuration])(implicit ec: ExecutionContext): Future[Value] = {
+  private def performRequestCommon[T](body: JsonNode, timeout: Option[FiniteDuration], handler: HttpResponse[String] => Future[T])(implicit ec: ExecutionContext): Future[T] = {
     val javaTimeout = timeout.map(_.toJava).asJava
     val response: Future[HttpResponse[String]] = connection.post("", body, javaTimeout).toScala
 
     response
       .flatMap {
-        case successResponse if successResponse.statusCode() < 300 => handleSuccessResponse(successResponse)
+        case successResponse if successResponse.statusCode() < 300 => handler.apply(successResponse)
```

Comment:
You could remove the `apply` and just write `handler(successResponse)` which is more idiomatic.

Reply:
yep, thanks
removed this method